src/generate.rs

```diff
@@ -244,7 +244,8 @@ pub fn interrupt(
         Target::CortexM => {
             let is_armv6 = match device.cpu {
                 Some(ref cpu) if cpu.name.starts_with("CM0") => true,
```

Comment:
Could the first two arms be merged into `Some(ref cpu) => cpu.starts_with("CM0")`?

And I think it would be better (for readability) to explicitly write `None` in the third arm.